core/app/models/spree/address.rb

```diff
@@ -177,6 +168,19 @@ def country_iso
       country && country.iso
     end
 
+    # @return [String] the full name on this address
+    def name
+      Spree::Name.new(firstname, lastname).full_name
+    end
+
+    def name=(value)
+      return if value.nil?
```

Comment:
I think that setting `firstname` and `lastname` to an empty string could make sense even if the model would be invalid. That guard condition was meant to avoid setting them to an empty string when `name` was forced to `nil` by a mass assignment with default values.